### ISSUE TITLE
Webhook creates to much CI/CD assets on GKE/AKS

### ISSUE THREAD

**Issue by czunker** [mondoohq/mondoo-operator#685]
**Describe the bug**
![image](https://user-images.githubusercontent.com/827818/201886243-f0358e2c-bc1d-4f7b-9dd4-d8420661caf1.png)

The webhook nearly constantly creates new CI/CD assets as seen in the screenshot above.
It looks like the assets aren't created from real changes to the k8s objects.

On first sight, it looks like only DaemonSets are affected.

**To Reproduce**
1. Deploy operator to AKS/GKE cluster.
2. Enable webhook.
3. Wait a short time.

**Expected behavior**
Webhook should only create CI/CD assets on changes to the k8s objects.

**Comment by czunker** [comment]
Looks like the addon manager is updating the `resourceVersion` every few seconds, and this triggers an `UPDATE` event for the webhook.

```
 ~  date; k -n kube-system get daemonset.apps/gke-metadata-server -o json | jq '.metadata.resourceVersion'                                                                                                                    Di 15. Nov 10:56:09 CET 2022
"294806637"
 ~  date; k -n kube-system get daemonset.apps/gke-metadata-server -o json | jq '.metadata.resourceVersion'                                                                                                                    Di 15. Nov 10:56:23 CET 2022
"294806943"
 ~  date; k -n kube-system get daemonset.apps/gke-metadata-server -o json | jq '.metadata.resourceVersion'                                                                                                                    Di 15. Nov 10:56:36 CET 2022
"294806943"
 ~  date; k -n kube-system get daemonset.apps/gke-metadata-server -o json | jq '.metadata.resourceVersion'                                                                                                                    Di 15. Nov 10:56:42 CET 2022
"294807133"
```

**Comment by czunker** [comment]
Default interval: `ADDON_CHECK_INTERVAL_SEC=60`

**Comment by czunker** [comment]
Can be partially reproduced locally with minikube and https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/addon-manager

**Comment by czunker** [comment]
Diffed two revisions of a DaemonSet and only the revision changed:
```
13c13
<   resourceVersion: "294966478"
---
>   resourceVersion: "294993038"
```

**Comment by czunker** [comment]
This is nothing specific to GKE, AKS or the addon-manager. It is a server-side apply bug: https://github.com/kubernetes/kubernetes/issues/95460

I can reproduce it with, e.g., a cert-manager deployment:
```
kubectl apply --force-conflicts --server-side=true -f https://github.com/cert-manager/cert-manager/releases/download/v1.8.0/cert-manager.yaml
kubectl -n cert-manager get deployments.apps/cert-manager -o json | jq '.metadata.resourceVersion'                                                                  
"255460"
kubectl apply --force-conflicts --server-side=true -f https://github.com/cert-manager/cert-manager/releases/download/v1.8.0/cert-manager.yaml
kubectl -n cert-manager get deployments.apps/cert-manager -o json | jq '.metadata.resourceVersion'                                                                 
"255508"
```
As I understand this PR, it is solved with k8s 1.25: https://github.com/kubernetes/kubernetes/pull/106388

So, seems we have to deal with it for some versions.
